src/puppetlabs/puppetdb/query/events.clj

```diff
@@ -47,6 +47,23 @@
     where)
    params])
 
+(defn with-latest-events
+  "CTE to wrap unioned queries when distinct_resources is used"
+  [query]
+  (format
+    "WITH latest_events AS
+           (SELECT certnames.certname,
+                   resource_events.resource_type,
+                   resource_events.resource_title,
+                   resource_events.property,
+                   MAX(resource_events.timestamp) AS timestamp
+                   FROM resource_events
+                   JOIN certnames ON resource_events.certname_id = certnames.id
```

Comment:
it's about bleedin' time. At some point we'll want to retrofit the rest of the queries that join on certname as a string.